mitiq/zne/scaling/folding.py

```diff
@@ -277,6 +277,12 @@ def fold_all(
         raise ValueError(
             f"Requires scale_factor >= 1 but scale_factor = {scale_factor}."
         )
+        
+    if scale_factor % 2 == 0:
```

Comment:
The `scale_factor` parameter seems to be a float, will the mod operator be sufficient here?